### PR TITLE
Adopt ASDInstallWebAttributionService replacing ASDInstallAttribution

### PR DESCRIPTION
**Pull request by achristensen07**
#### b40525e847936bf95df365170dfbd4a9175e0e9b
<pre>
Adopt ASDInstallWebAttributionService replacing ASDInstallAttribution
<a href="https://bugs.webkit.org/show_bug.cgi?id=257006">https://bugs.webkit.org/show_bug.cgi?id=257006</a>
rdar://100043903

Reviewed by John Wilander.

And adopt the new entitlement.  Remove the code using ASDInstallAttribution
on platforms that don&apos;t have ASDInstallWebAttributionService because we
don&apos;t need to keep it working on those platforms because those platforms
will never see this change.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebKit/NetworkProcess/cocoa/AppStoreDaemonSPI.h:
* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm:
(WebKit::NetworkSessionCocoa::donateToSKAdNetwork):
* Source/WebKit/Scripts/process-entitlements.sh:

Canonical link: <a href="https://commits.webkit.org/264324@main">https://commits.webkit.org/264324@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/d0dbf189095f0e3723b6f0ec02dd7f8556af8539

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/7192 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7440 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7619 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8809 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7412 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/8771 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7373 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/10315 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7319 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/8016 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/6630 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8916 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/7304 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/5368 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/6549 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/14284 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/6102 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/7012 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/6650 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/9526 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/6777 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7128 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/5829 "1 failures") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/7331 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6488 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1691 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10687 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/7533 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/860 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6872 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/1831 "Passed tests") |

### CODE REVIEW

File: Source/WebKit/NetworkProcess/cocoa/AppStoreDaemonSPI.h [hunk 1]
```diff
@@ -45,10 +48,12 @@ typedef NS_ENUM(NSInteger, ASDInstallWebAttributionContext) {
 @property (nonatomic, assign) ASDInstallWebAttributionContext attributionContext;
 @end
 
-@interface ASDInstallAttribution : NSObject
-+ (instancetype)sharedInstance;
-- (void)addInstallWebAttributionParamsWithConfig:(ASDInstallWebAttributionParamsConfig *)config completionHandler:(void (^)(NSError *error))completionHandler;
+#if HAVE(ASD_INSTALL_WEB_ATTRIBUTION_SERVICE)
```
Comment (johnwilander) [inline Source/WebKit/NetworkProcess/cocoa/AppStoreDaemonSPI.h:51]: Won't we regress on platforms that don't have this service?

Comment (achristensen07) [reply Source/WebKit/NetworkProcess/cocoa/AppStoreDaemonSPI.h:51]: That platform is iOS 16, which will never see this change.

Comment (johnwilander) [reply Source/WebKit/NetworkProcess/cocoa/AppStoreDaemonSPI.h:51]: Thanks!

**Review by johnwilander — APPROVED** [review]
r=me. The Mac-AS-debug-wk2 test failure looks unrelated.

**Pull request merged**